tests/primitive_list.cpp

```diff
@@ -0,0 +1,799 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include "catch.hpp"
+
+#include "util/event_loop.hpp"
+#include "util/index_helpers.hpp"
+#include "util/templated_test_case.hpp"
+#include "util/test_file.hpp"
+
+#include "binding_context.hpp"
+#include "list.hpp"
+#include "object.hpp"
+#include "object_schema.hpp"
+#include "property.hpp"
+#include "results.hpp"
+#include "schema.hpp"
+#include "thread_safe_reference.hpp"
+
+#include "impl/realm_coordinator.hpp"
+#include "impl/object_accessor_impl.hpp"
+
+#include <realm/group_shared.hpp>
+#include <realm/link_view.hpp>
+#include <realm/query_expression.hpp>
+#include <realm/version.hpp>
+
+#include <numeric>
+
+using namespace realm;
+
+template<PropertyType prop_type, typename T>
+struct Base {
+    using Type = T;
+    using Wrapped = T;
+    using Boxed = T;
+
+    static PropertyType property_type() { return prop_type; }
+    static std::vector<int64_t> values() { return {3, 1, 2}; }
```

Comment:
It looks like the expectation is that each subclass provides an implementation of `values()`. This base class version is unused, and wouldn't work for most subclasses anyway. Probably best to just omit it?